pkg/addon/controllers/addonconfiguration/graph.go

```diff
@@ -461,3 +495,53 @@ func rolloutStatusHasCluster(clusterRolloutStatus []clustersdkv1alpha1.ClusterRo
 	}
 	return false
 }
+
+func overrideConfigMapByInstallConfigRef(
+	installConfigRefs []addonv1alpha1.InstallConfigReference,
+	desiredConfigs addonConfigMap,
+) {
+	gvkOverwritten := make(map[addonv1alpha1.ConfigGroupResource]struct{})
+	for _, configRef := range installConfigRefs {
+		if configRef.DesiredConfig == nil {
+			continue
+		}
+		gr := configRef.ConfigGroupResource
+		referent := configRef.DesiredConfig.ConfigReferent
+		if _, exist := gvkOverwritten[gr]; !exist {
+			desiredConfigs[gr] = []addonv1alpha1.ConfigReference{}
+			gvkOverwritten[gr] = struct{}{}
+		}
+		if _, exist := desiredConfigs.containsConfig(gr, referent); !exist {
+			desiredConfigs[gr] = append(desiredConfigs[gr], addonv1alpha1.ConfigReference{
+				ConfigGroupResource: gr,
+				ConfigReferent:      referent,
+				DesiredConfig:       configRef.DesiredConfig.DeepCopy(),
+			})
+		}
+	}
+}
+
+func overrideConfigMapByAddOnConfigs(
```

Comment:
ditto we need more docs for matinability

Reply:
added